source/common/protobuf/utility.cc

```diff
@@ -426,7 +426,10 @@ std::string MessageUtil::getJsonStringFromMessage(const Protobuf::Message& messa
   std::string json;
   const auto status = Protobuf::util::MessageToJsonString(message, &json, json_options);
   // This should always succeed unless something crash-worthy such as out-of-memory.
-  RELEASE_ASSERT(status.ok(), "");
+  if (!status.ok()) {
```

Comment:
It seems like the comment above needs to be adjusted? Why is this failing? Shouldn't we always be able to covert a message to a JSON string?

Reply:
Yep, in this case this was an OOM. The message was a protobuf struct that was recursive. (The mutator that produces these test-cases sets a recursive protobuf limit of 100, but this was still enough to OOM). Seems like the same happens in `jsonConvertInternal`

Reply:
Hmm, shouldn't we actually crash then? Or it's not an OOM persay but some kind of max recursion on the conversion? The main thing I'm worried about here is I'm not sure it's safe to throw an exception in all cases where this code is used?

Reply:
Hmm yeah. It looks like admin endpoint uses this. In the access log case, the string returned from this method is truncated (but only after the failure).

It also seems like the filter metadata is a trusted source, and no one will be overloading it with recursive protobuf structs, so I moved the truncation to stop the failure in to the utility functions for the fuzzer. Harvey also opened this issue: https://github.com/envoyproxy/protoc-gen-validate/issues/95